test/xpu_api/numerics/numeric.ops/numeric.ops.gcd/xpu_gcd.pass.cpp

```diff
@@ -0,0 +1,135 @@
+                using S2 = oneapi::dpl::make_signed_t<Input2>;
+                using U1 = oneapi::dpl::make_unsigned_t<Input1>;
+                using U2 = oneapi::dpl::make_unsigned_t<Input2>;
+                constexpr struct
+                {
+                    int x;
+                    int y;
+                    int expect;
+                } Cases[] = {{0, 0, 0}, {1, 0, 1}, {0, 1, 1},   {1, 1, 1},
+                             {2, 3, 1}, {2, 4, 2}, {36, 17, 1}, {36, 18, 18}};
+                for (auto TC : Cases)
+                {
+                    { // Test with two signed types
+                        using Output = oneapi::dpl::common_type_t<S1, S2>;
+                        out[0] &= test0<S1, S2, Output>(TC.x, TC.y, TC.expect);
+                        out[0] &= test0<S1, S2, Output>(-TC.x, TC.y, TC.expect);
+                        out[0] &= test0<S1, S2, Output>(TC.x, -TC.y, TC.expect);
+                        out[0] &= test0<S1, S2, Output>(-TC.x, -TC.y, TC.expect);
+                        out[0] &= test0<S2, S1, Output>(TC.x, TC.y, TC.expect);
+                        out[0] &= test0<S2, S1, Output>(-TC.x, TC.y, TC.expect);
+                        out[0] &= test0<S2, S1, Output>(TC.x, -TC.y, TC.expect);
+                        out[0] &= test0<S2, S1, Output>(-TC.x, -TC.y, TC.expect);
+                    }
+
+                    { // test with two unsigned types
+                        using Output = oneapi::dpl::common_type_t<U1, U2>;
+                        out[0] &= test0<U1, U2, Output>(TC.x, TC.y, TC.expect);
+                        out[0] &= test0<U2, U1, Output>(TC.x, TC.y, TC.expect);
+                    }
+                    { // Test with mixed signs
+                        using Output = oneapi::dpl::common_type_t<S1, U2>;
+                        out[0] &= test0<S1, U2, Output>(TC.x, TC.y, TC.expect);
+                        out[0] &= test0<U2, S1, Output>(TC.x, TC.y, TC.expect);
+                        out[0] &= test0<U2, S1, Output>(TC.x, -TC.y, TC.expect);
+                        out[0] &= test0<S1, U2, Output>(-TC.x, TC.y, TC.expect);
+                    }
+                    { // Test with mixed signs
+                        using Output = oneapi::dpl::common_type_t<S2, U1>;
+                        out[0] &= test0<S2, U1, Output>(TC.x, TC.y, TC.expect);
+                        out[0] &= test0<U1, S2, Output>(TC.x, TC.y, TC.expect);
+                        out[0] &= test0<U1, S2, Output>(TC.x, -TC.y, TC.expect);
+                        out[0] &= test0<S2, U1, Output>(-TC.x, TC.y, TC.expect);
+                    }
+                }
+                {
+                    auto res = oneapi::dpl::gcd(static_cast<std::int64_t>(1234), INT32_MIN);
+                    out[0] &= (res == 2);
+                }
+            });
+        });
+    }
+
+    if (res)
+        std::cout << "pass\n";
+    else
+        std::cout << "fail\n";
```

Comment:
I believe we don't want to have one print per call. What we do want to have is assert over `res`. I would suggest to change `if-else` statement with output to `assert(res)`

Reply:
Done.

Reply:
@XunZhangIntel 
It remembered automation test replied on test case's output, if we use "assert" instead of "if-else" here, can you handle?

Reply:
@jinge90 Yes, it can be handled.